include/exadg/structure/preconditioners/multigrid_preconditioner.cpp

```diff
@@ -122,7 +122,7 @@ MultigridPreconditioner<dim, Number>::update()
 
   // In case that the operators have been updated, we also need to update the smoothers and the
   // coarse grid solver. This is generic functionality implemented in the base class.
-  if(nonlinear or data.unsteady)
+  if(nonlinear or data.unsteady or this->update_needed)
```

Comment:
I agree that in the linear steady case we currently never run into this part of the code, which is a problem. However, this if-statement looks somewhat irritating to me. Isn't the consequence of what you found out that we should remove the if-statement completely?  In which cases, where we run into `preconditioner->update()`, do we not want to update the smoother/coarse solver. When comparing e.g. with the ConvDiff MG preconditioner, we do not have any if-statement before these update-calls? Could you please check this.

Reply:
I will check soon, if `(nonlinear or data.unsteady or this->update_needed)` actually ever evaluates to `false` and then push the new version.

Reply:
the condition is now completely removed, since it always gave true for any of the combinations tested (Steady, Unsteady, QuasiStatic with finite or infinitesimal strains (and AMG, PointJacobi and Multigrid preconditioners)).